benchmark/scenarios/test/test_write_request_load.py

```diff
@@ -0,0 +1,362 @@
+from ipaddr import IPAddress
+from twisted.internet.defer import succeed, Deferred
+from twisted.internet.task import Clock
+from twisted.python.components import proxyForInterface
+from twisted.trial.unittest import SynchronousTestCase
+
+from flocker.apiclient._client import (
+    IFlockerAPIV1Client, FakeFlockerClient, Node
+)
+
+from benchmark.scenarios import (
+    WriteRequestLoadScenario, WRateMeasurer, WRequestRateTooLow,
+    WRequestRateNotReached, WRequestOverload, WDataseCreationTimeout
+)
+
+from benchmark.scenarios.read_request_load import DEFAULT_SAMPLE_SIZE
+
+from benchmark.cluster import BenchmarkCluster
+
+
+class WRateMeasurerTest(SynchronousTestCase):
+    """
+    WRateMeasurer tests
+    """
+
+    def send_requests(self, r_measurer, num_req, num_samples):
+        """
+        Helper function that will send the desired number of request.
+
+        :param r_measurer: `rate_measurer` we are testing
+        :param num_req: number of request we want to send.
+        :param num_samples: numbe of samples (interval)
+        """
+        for i in range(num_samples):
+            for i in range(num_req):
+                r_measurer.send_request()
+
+    def receive_requests(self, r_measurer, num_req, num_samples):
+        """
+        Helper function that will receive the desired number of requests.
+
+        :param r_measurer: `rate_measurer` we are testing
+        :param num_req: number of request we want to receive.
+        :param num_samples: numbe of samples (interval)
+        """
+        ignored = ""
+        for i in range(num_samples):
+            for i in range(num_req):
+                r_measurer.receive_request(ignored)
+            r_measurer.update_rate()
+
+    def increase_rate(self, r_measurer, num_req, num_samples):
+        """
+        Helper function that will increase the rate, sending the
+        desired number of request, and receiving the same
+        amount of them.
+
+        :param r_measurer: `rate_measurer` we are testing
+        :param num_req: number of request we want to make.
+        :param num_samples: numbe of samples (interval)
+        """
+        self.send_requests(r_measurer, num_req, num_samples)
+        self.receive_requests(r_measurer, num_req, num_samples)
+
+    def test_rate_is_zero_when_no_samples(self):
+        """
+        When no samples have been collected, the rate should be 0.
+        """
+        r = WRateMeasurer()
+        self.assertEqual(r.rate(), 0, "Expected initial rate to be zero")
+
+    def test_rate_is_small_when_not_enough_samples(self):
+        """
+        When the number of samples collected is less than the sample
+        size, the rate should be smaller than `req_per_second`.
+        """
+        r = WRateMeasurer()
+        req_per_second = 5
+
+        self.increase_rate(r, req_per_second, (r.sample_size / 2))
+
+        self.assertEqual((req_per_second * (r.sample_size / 2)) /
+                         r.sample_size,
```

Comment:
It might make sense to use an alternative line wrapping style here:
```.py
    self.assertEqual(
        (req_per_second * (r.sample_size / 2)) / r.sample_size,
        ...,
    )
```

It's probably easier to parse the long expression if it is on one line.